### ISSUE TITLE
make warp (banish-cursorwarp) work with the togglescratch function (renamedscratchpads)

### ISSUE THREAD

**Issue by adetabrani** [bakkeby/patches#75]
[simplescreenrecorder.webm](https://github.com/bakkeby/patches/assets/47938004/de9bf4b4-1ebe-4a3e-bdcc-11d77c574e1b)

hi bak, how to make warp work with the togglescratch function? I tried adding the 'warp(c)' function call inside the togglescratch function, but it always ended up crashing 😂

**Comment by bakkeby** [comment]
Presumably you only want to call `warp` when the client is revealed?

**Comment by adetabrani** [comment]
yes, because of the current behavior I have to always refocus when the scratchpad window is toggled.

I want when I toggle scratchpad, the cursor automatically move to the scratchpad window, but I'm not sure what if there is more than 1 scratchpad window being managed, where should the cursor be located?

**Comment by adetabrani** [comment]
is there any feedback regarding this issue?

**Comment by bakkeby** [comment]
Have you given any more thought into your last question?

> I'm not sure what if there is more than 1 scratchpad window being managed, where should the cursor be located?

If it were up to you and there were two scratchpad windows toggled together - where would you want the mouse cursor to warp to (if at all)?

**Comment by adetabrani** [comment]
hmmm, maybe for the last window that is set as a scratchpad, to be honest, I often use just one window as a scratchpad

**Comment by bakkeby** [comment]
Have you considered how the warp patch works for non-scratchpad clients? Does the mouse cursor stay with e.g. the last window spawned?

**Comment by adetabrani** [comment]
ah I almost forgot about that, maybe instead we can warp to the first stack on the scratchpad window (if more than 1 scratchpad window)? honestly I rarely set more than 1 scratchpad window, so I think at least warp should be able to work first for one scratchpad window
